src/Json.Schema.ToDotNet.UnitTests/DataModelGeneratorTests.cs

```diff
@@ -2688,13 +2737,43 @@ public virtual object VisitActual(ISNode node)
         }
 
         private T VisitNullChecked<T>(T node) where T : class, ISNode
+        {
+            string emptyKey = null;
+            return VisitNullChecked<T>(node, ref emptyKey);
+        }
+
+        private T VisitNullChecked<T>(T node, ref string key) where T : class, ISNode
         {
             if (node == null)
             {
                 return null;
             }
 
-            return (T)Visit(node);
+            if (key == null)
+            {
+                return (T)Visit(node);
+            }
+
+            return (T)VisitDictionaryEntry(node, ref key);
+        }
+
+        private ISNode VisitDictionaryEntry(ISNode node, ref string key)
```

Comment:
>VisitDictionaryEntry [](start = 23, length = 20)

If no schema properties are defined as dictionaries of schema-defined types (as in this test), there's no reason to emit this method or the two-argument overload to `VisitNullChecked`. I suggest we not further complicate the code generator by optimizing for this.